### ISSUE TITLE
Best practice of putting values into a nested json using TBEL

### ISSUE THREAD

**Issue by ZYMCao** [thingsboard/thingsboard#12280]
**Component**

 * Rule Engine

**Description**
I have a rather common requirement while using the script node of the rule engine: 
given a list being ["A", "B", "C", "D"] and a value being "x"; and a nested json. I want to insert value "x" into the nested json according to the list. The thing is this nested json does not necessarily look like 
var json = {
    A: {
        B: {
            C: {
                D: "y"
            }
        }
    }
}
ready for me to insert "x" to node D. It may lack node D, it may lack node C, it may be an empty {}...
Using put method here is already lengthy and error-prone; in addition, A node may have other fields in addition to B, the same applies to node B, C, and D (if they exist) in which case I can't put values without getting value from this nested json first.

I know to meet my requirement, basic get method and put method provided by TBEL for json can work; but somehow I feel like there is a better way out there.

PS: I feel like I am looking for something similar to the inverse functionality of JSONPath

**Comment by epalchenko** [comment]
Hello,

Please provide "Payload"(message), “Metadata, “ and "function Transform" in .json format for a more in-depth investigation.

**Comment by ZYMCao** [comment]
NVM, I fixed it (kinda) with the following function "
function nodeSetter(json, list, value) {
    var currentNode = json;
    for (var i = 0; i < list.length - 1; i++) {
        var key = list[i];
        
        if (currentNode.get(key) == null) {
            currentNode.put(key, {});
        }
        
        currentNode = currentNode.get(key);
    }

    currentNode.put(list[list.length - 1], value);
    return json;
}
".

Basically nodeSetter({}, ["A", "B", "C", "D"], "x") will give me a json of "{A: {B: {C: {D: "x"}}}}" which is what I need.
The main struggle I had was reference assignment of the first line, "var currentNode = json", whose implication seems to be different from java.
